app/collections/settings.jsx

```diff
@@ -84,6 +89,37 @@ export default class CollectionSettings extends React.Component {
       });
   }
 
+  handleDescriptionInputChange() {
+    const property = 'description';
+    const setting = this.state.setting;
+    setting[property] = true;
+    this.setState({ error: null, setting });
+
+    const descriptionText = this.description.value;
+    const changes = {};
+    changes[property] = descriptionText;
+
+    if (descriptionText.length > 300) {
+      this.setState({
+        descriptionLengthError: true
+      });
+      return;
+    } else {
+      this.setState({
+        descriptionLengthError: false
+      });
+    }
+
+    this.props.collection.update(changes).save()
```

Comment:
The AutoSave component is bad UX most of the places we've used it. There's a refactored DisplayNameSlugEditor that doesn't use the AutoSave component in ZRC. But we're not quite ready to add ZRC here yet. In the meantime, then to be consistent with behavior, using the AutoSave component for the description is ok. 

Now, since we're using the AutoSave component, it handles the update and save of the resource for you, so these lines are duplicating the request.